.github/workflows/build_workflow.yml

```diff
@@ -107,11 +107,17 @@ jobs:
         uses: gradle/actions/setup-gradle@v4
 
       - name: Execute Unit Tests
+        if: github.actor != 'dependabot[bot]'
```

Comment:
I'm not sure about this. We surely want some confidence that a dependency bump doesn't break CI?

Skipping mutation testing makes sense, as dependabot doesn't change our production code for which the mutations would be generated for.

Skipping our unit tests is scary though.

Reply:
Just seen the addition below. Is there a way to combine the two jobs? Conditionally add 'sonar' to the list of jobs depending on the actor.

Reply:
Yeah,  I wouldn't have skipped the unit tests!

I think that the change I've made should work now with the two conditional steps moved into one and using a conditional in the run script